### PR TITLE
Fix selecting files from folders in `FileField`

### PR DESCRIPTION
**Pull request by thomasdax98**
## Problem

`FileField` multi-select silently dropped any selected file that lived in a subfolder. `damFilesList` implicitly constrains to `folder.id = null` (scope root) when no `folderId` is passed, so resolving the selection via `filter: { ids: ... }` filtered out everything outside the root. An escape hatch already existed for `filter.searchText`, but not for `filter.ids`.

## Solution

Bypass the implicit folder constraint when `filter.ids` is set — mirroring the existing `searchText` behavior. Folder-browsing (default-to-root, explicit `folderId`) is unchanged.

The behavior is now documented on `FileFilterInput.searchText`, `FileFilterInput.ids`, and `FileArgs.folderId`.

## Tests

`files.service.spec.ts` covers 6 cases × `findAll` / `findAndCount` (12 tests): defaults, explicit `folderId`, the new `ids` bypass, `folderId` + `ids` together, empty `ids` array, and a regression guard for `searchText`.

### CODE REVIEW

File: packages/api/cms-api/src/dam/files/files.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+const callers = [
+    {
+        name: "findAll",
+        call: (service: FilesService, args: CallArgs) => service.findAll(args),
+    },
+    {
+        name: "findAndCount",
+        call: (service: FilesService, args: CallArgs) => service.findAndCount({ ...args, offset: 0, limit: 25 }),
+    },
+];
```
Comment (thomasdax98) [inline packages/api/cms-api/src/dam/files/files.service.spec.ts:64]: This pattern was Claude's idea. I was sceptical at first, but actually it makes a lot of sense because `findAll` and `findAndCount` should always behave identical regarding filtering and so on

**Review by johnnyomair — APPROVED** [review]
